tests/test_package_specifier.py

```diff
@@ -64,3 +67,60 @@ def test_parse_specifier_for_metadata(
         print(f"package_spec_in = {package_spec_in}")
         with pytest.raises(PipxError, match=r"^Unable to parse package spec"):
             package_or_url = parse_specifier_for_metadata(package_spec_in)
+
+
+@pytest.mark.parametrize(
+    "package_spec_in,pip_args_in,package_spec_expected,pip_args_expected,warning_str",
+    [
+        ('pipx==0.15.0;python_version>="3.6"', [], "pipx==0.15.0", [], None),
+        ("pipx==0.15.0", ["--editable"], "pipx==0.15.0", [], "Ignoring --editable",),
+        (
+            'pipx==0.15.0;python_version>="3.6"',
+            [],
+            "pipx==0.15.0",
+            [],
+            "Ignoring environment markers",
```

Comment:
Nit: Could the environment markers be captured and printed?

Reply:
Do you mean specifically in the test?  Or do you mean generally for the `pipx` warnings to the user?

Reply:
Generally for user facing earnings, which would mean the test would also need to be updated. Just a nit, I’m fine merging without the change as well.

Reply:
It sounds like a good idea.  I'll implement it and then merge.

Reply:
Looks good. One other thing just came to mind. If I were a user and got that message, I would probably want to know *why* things were being ignored. There are good reasons but they are not always obvious (as your discussion in the related issue revealed). Could reasoning also be added to the warning when arguments are being dropped by pipx?

Reply:
I can imagine saying for the environment marker case: "Use pipx options instead." as concise but still basically explaining the philosophy.  Does that sound good?

I'm a little at a loss at a concise (warning message length) way to explain ignoring `--editable` for non-local path.  I'm thinking.  Anyone have an idea?

Reply:
Sorry for the long delay. The text you ended up with is perfect.